src/modules/rc_update/rc_update.cpp

```diff
@@ -685,10 +685,10 @@ void RCUpdate::UpdateManualControlInput(const hrt_abstime &timestamp_sample)
 	manual_control_input.data_source = manual_control_setpoint_s::SOURCE_RC;
 
 	// limit controls
-	manual_control_input.y     = get_rc_value(rc_channels_s::FUNCTION_ROLL,    -1.f, 1.f);
-	manual_control_input.x     = get_rc_value(rc_channels_s::FUNCTION_PITCH,   -1.f, 1.f);
-	manual_control_input.r     = get_rc_value(rc_channels_s::FUNCTION_YAW,     -1.f, 1.f);
-	manual_control_input.z     = get_rc_value(rc_channels_s::FUNCTION_THROTTLE, -1.f, 1.f);
+	manual_control_input.roll = get_rc_value(rc_channels_s::FUNCTION_ROLL,    -1.f, 1.f);
+	manual_control_input.pitch = get_rc_value(rc_channels_s::FUNCTION_PITCH,   -1.f, 1.f);
+	manual_control_input.yaw = get_rc_value(rc_channels_s::FUNCTION_YAW,     -1.f, 1.f);
+	manual_control_input.throttle = get_rc_value(rc_channels_s::FUNCTION_THROTTLE, -1.f, 1.f);
```

Comment:
Ok then it seems like this is the place where we need to tackle the case for the throttle's trim being set to minimum.

And that is handled here:
https://github.com/PX4/PX4-Autopilot/blob/45b390b0bfe21fc591b36ec898f6fc0791fc4729/src/modules/rc_update/rc_update.cpp#L454-L482

How could we improve this?

Reply:
And it seems like the comment `TRIM needs to be set to Minimum, for throttle channel` is everywhere in the Params.c file 🤣. We should remove this!

https://github.com/PX4/PX4-Autopilot/blob/45b390b0bfe21fc591b36ec898f6fc0791fc4729/src/modules/rc_update/params.c#L53-L63

Reply:
Taken from a random log: https://logs.px4.io/plot_app?log=57806d35-3c0f-4319-99d2-2da69d22a8dc

![image](https://user-images.githubusercontent.com/23277211/202847037-b4d47044-e866-43e1-a64a-ca4c7ebd2768.png)

Now I am spotting all these TRIM set to MIN evils everywhere 🙈!

❗ I think that the way TRIM value is affecting the translation of the RC raw value into RC input values is the problem.

Especially:
```
First normalize to 0..1 range with correct sign (below or above center), the total range is 2 (-1..1).

If center (trim) == min, scale to 0..1, if center (trim) == max, scale to -1..0.
```

I feel like `rc_update.cpp` shouldn't mess with the actual output range of the translated channels normalized value, dependin on TRIM setting (on min or max value, which is an edge case).

The main questions are:
1. Why do we want to use the 'trim' value to configure output to [0, 1] or [-1 ,0] range, what are the use-cases for this?
2. If there is a use-case, and this trim setting method makes sense, how can we somehow re-enable the full range [-1, 1] for the `manual_control_setpoint.throttle`? Do we need to handle this case specially?

We could bring this up in the dev-call 🤔 @MaEtUgR

Reply:
I mean, one quick workaround that comes to my mind is to just check if we are dealing with a throttle channel, and if so, don't apply the trim value if it is at min or max value, and just do the mapping between min ~ max as -1 ~ +1

Reply:
Solved in https://github.com/PX4/PX4-Autopilot/commit/d5e2573434cdf0799900a8fdacffe29884f5f510

Reply:
By the way thanks for the hint about the parameter metadata comments. I removed them in https://github.com/PX4/PX4-Autopilot/commit/92510c6906ee5593de0e229d43b0c1fdebff43ea

The rc update function applying the calibration I plan to refactor as a follow up.

Reply:
@junwoo091400 Here's my refactor of the RC update function applying the calibration: https://github.com/PX4/PX4-Autopilot/pull/20648